pkg/client/client_test.go

```diff
@@ -1 +1,89 @@
 package client
+
+import (
+	"io/ioutil"
+	"os"
+	"testing"
+
+	"github.com/aliyun/alibaba-cloud-sdk-go/sdk/auth"
+
+	"github.com/stretchr/testify/assert"
+
+	"github.com/aliyun/alibaba-cloud-sdk-go/sdk"
+	"github.com/aliyun/alibaba-cloud-sdk-go/sdk/auth/credentials/provider"
+	sdkerrors "github.com/aliyun/alibaba-cloud-sdk-go/sdk/errors"
+	"github.com/aliyun/alibaba-cloud-sdk-go/services/ecs"
+)
+
+var (
+	testCreads = `
+	[default]
+	type = access_key
+	access_key_id = YourAccessKeyId
+	access_key_secret = YourAccessKeySecret
+	`
+)
+
+func getCredentials() (auth.Credential, error) {
+	creds := []byte(testCreads)
+	f, err := ioutil.TempFile("", "alibaba-creds-*")
+	if err != nil {
+		return nil, err
+	}
+
+	defer os.Remove(f.Name())
+	defer f.Close()
+
+	_, err = f.Write(creds)
+	if err != nil {
+		return nil, err
+	}
+
+	os.Setenv(provider.ENVCredentialFile, f.Name())
+	defer os.Unsetenv(provider.ENVCredentialFile)
+	// use Alibaba provider initialization
+	p := provider.NewProfileProvider("default")
+	// return a valid auth credential
+	return p.Resolve()
+}
+
+func Test_EcsEndPoint(t *testing.T) {
+	credential, err := getCredentials()
+	assert.Nil(t, err)
+
+	sdkConfig := &sdk.Config{
+		UserAgent: machineProviderUserAgent,
+		Scheme:    "HTTPS",
+	}
+
+	endpointMap := ecs.GetEndpointMap()
+	endpointMap["cn-zhangjiakou"] = "ecs.cn-zhangjiakou.aliyuncs.com"
+	endpointMap["cn-huhehaote"] = "ecs.cn-huhehaote.aliyuncs.com"
+	endpointMap["cn-wulanchabu"] = "ecs.cn-wulanchabu.aliyuncs.com"
+	endpointMap["cn-heyuan"] = "ecs.cn-heyuan.aliyuncs.com"
+	endpointMap["cn-guangzhou"] = "ecs.cn-guangzhou.aliyuncs.com"
+	endpointMap["cn-chengdu"] = "ecs.cn-chengdu.aliyuncs.com"
+	endpointMap["ap-northeast-1"] = "ecs.ap-northeast-1.aliyuncs.com"
+	endpointMap["ap-southeast-2"] = "ecs.ap-southeast-2.aliyuncs.com"
+	endpointMap["ap-southeast-3"] = "ecs.ap-southeast-3.aliyuncs.com"
+	endpointMap["ap-southeast-5"] = "ecs.ap-southeast-5.aliyuncs.com"
+	endpointMap["ap-southeast-6"] = "ecs.ap-southeast-6.aliyuncs.com"
+	endpointMap["ap-south-1"] = "ecs.ap-south-1.aliyuncs.com"
+	endpointMap["eu-west-1"] = "ecs.eu-west-1.aliyuncs.com"
+	endpointMap["me-east-1"] = "ecs.me-east-1.aliyuncs.com"
+	endpointMap["eu-central-1"] = "ecs.eu-central-1.aliyuncs.com"
+	endpointMap["cn-nanjing"] = "ecs.cn-nanjing.aliyuncs.com"
```

Comment:
Is there an exhaustive list of these somewhere we could reference to make sure have all of the possibilities? Maybe there are constants in the SDK we can leverage here?

Reply:
Most of our cloud products have an API named DescribeRegions.  Here is the ECS api documentation. https://www.alibabacloud.com/help/en/doc-detail/25609.htm?spm=a3c0i.23458820.2359477120.4.4c6f7d3fhtz9x6

Reply:
The SDK selects the correct endpoint by default. However, the ECS TEAM updated the endpoint earlier and did not update the configuration in the SDK. Therefore, when us-east-1 region is selected, an endpoint such as ecs.cn-hangzhou.aliyuncs.com is returned